app/src/main/java/com/github/houseorganizer/houseorganizer/MainActivity.java

```diff
@@ -6,27 +6,30 @@
 
 import androidx.appcompat.app.AppCompatActivity;
 
+import com.github.houseorganizer.houseorganizer.login.LoginActivity;
+import com.github.houseorganizer.houseorganizer.login.LoginEmail;
 import com.google.firebase.auth.FirebaseAuth;
 import com.google.firebase.auth.FirebaseUser;
 
 public class MainActivity extends AppCompatActivity {
 
-    private FirebaseAuth mAuth;
-
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
 
-        mAuth = FirebaseAuth.getInstance();
+        FirebaseAuth mAuth = FirebaseAuth.getInstance();
         FirebaseUser user = mAuth.getCurrentUser();
 
         /* If user is not authenticated, send him to LoginActivity to authenticate first.
-        * Else send him to MainScreenActivity */
+         * Else send him to MainScreenActivity */
         (new Handler()).postDelayed(() -> {
             if (user != null) {
-                Intent mainScreenIntent = new Intent(this, MainScreenActivity.class);
-                startActivity(mainScreenIntent);
+                if (user.isEmailVerified()) {
```

Comment:
Maybe add one more check here: `if (user.isEmailVerified() || user.isAnonymous()))` s.t. anon users aren't redirected to LoginEmail

Reply:
good thinking i added it because its required but for now it makes the app crash because anonymous are not in houses

Reply:
Ok, let's mark it as a bug to fix in one of the upcoming sprints